lib/full-site-editing/class-wp-block-template.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Blocks API: WP_Block_Template class
+ *
+ * @package WordPress
+ * @since 5.5.0
+ */
+
+/**
+ * Class representing a block template.
+ */
+class WP_Block_Template {
+
+	/**
+	 * Type: wp_template or wp_template_part
+	 *
+	 * @var string
+	 */
+	public $type;
```

Comment:
Should we maybe differentiate these more -- e.g. by adding a separate `WP_Block_Template_Part` class? After all, they are quite different in some regards: e.g. templates are subject to template resolution, while template parts aren't. Different classes might help with better mental models and less confusion between templates and template parts.

Reply:
Aside, more lines of code, I don't think it's necessary at this point personally, I'm open to it for later though. The logic is exactly the same for everything in this PR.